test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -281,6 +286,84 @@ func clusterUpgradeWithRuntimeSDKSpec(ctx context.Context, inputGetter func() cl
 	})
 }
 
+// machineSetPreflightChecksTestHandler verifies the MachineSet preflight checks.
+// At this point in the test the ControlPlane is upgraded to the new version and the upgrade to the MachineDeployments
+// should be blocked by the AfterControlPlaneUpgrade hook.
+// Test the MachineSet preflight checks by scaling up the MachineDeployment. The creation on the new Machine
+// should be blocked because the preflight checks should not pass (kubeadm version skew preflight check should fail).
+func machineSetPreflightChecksTestHandler(ctx context.Context, c client.Client, clusterRef types.NamespacedName) {
+	// Verify that the hook is called and the topology reconciliation is blocked.
+	hookName := "AfterControlPlaneUpgrade"
+	Eventually(func() error {
+		if err := checkLifecycleHooksCalledAtLeastOnce(ctx, c, clusterRef, []string{hookName}); err != nil {
+			return err
+		}
+
+		cluster := framework.GetClusterByName(ctx, framework.GetClusterByNameInput{
+			Name: clusterRef.Name, Namespace: clusterRef.Namespace, Getter: c})
+
+		if !clusterConditionShowsHookBlocking(cluster, hookName) {
+			return errors.Errorf("Blocking condition for %s not found on Cluster object", hookName)
+		}
+
+		return nil
+	}, 30*time.Second).Should(Succeed(), "%s has not been called", hookName)
+
+	// Scale up the MachineDeployment
+	machineDeployments := framework.GetMachineDeploymentsByCluster(ctx, framework.GetMachineDeploymentsByClusterInput{
+		Lister:      c,
+		ClusterName: clusterRef.Name,
+		Namespace:   clusterRef.Namespace,
+	})
+	md := machineDeployments[0]
+
+	patchHelper, err := patch.NewHelper(md, c)
+	Expect(err).To(BeNil())
+
+	// Scale up the MachineDeployment.
+	// Since the MachineDeployment is pending an upgrade at this point the topology controller will not push any changes
+	// to the MachineDeployment. Therefore, the changes made to the MachineDeployment here will not be replaced
+	// until the AfterControlPlaneUpgrade hook unblocks the upgrade.
+	*md.Spec.Replicas++
+	Eventually(func() error {
+		return patchHelper.Patch(ctx, md)
+	}).Should(Succeed(), "Failed to scale up the MachineDeployment %s", klog.KObj(md))
+	// Verify the MachineDeployment updated replicas are not overridden.
+	// Note: This verifies that the topology controller in fact holds any reconciliation of this MachineDeployment.
+	Consistently(func(g Gomega) {
+		// Get the updated MachineDeployment.
+		targetMD := &clusterv1.MachineDeployment{}
+		// Wrap in an Eventually block for additional safety. Since all of this is in a Consistently block it
+		// will fail if we hit a transient error like a network flake.
+		Eventually(func() error {
+			return c.Get(ctx, client.ObjectKeyFromObject(md), targetMD)
+		}).Should(Succeed(), "Failed to get MachineDeployment %s", klog.KObj(md))
+		// Verify replicas are not overridden.
+		g.Expect(targetMD.Spec.Replicas).To(Equal(md.Spec.Replicas))
+	}, "10s", "1s")
```

Comment:
```suggestion
	}, 10*time.Second, 1*time.Second)
```
nit: let's use durations directly as we usually do (Just for consistency)